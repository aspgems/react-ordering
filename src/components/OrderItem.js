import React, { Component } from 'react';
import {
  ListGroupItemHeading,
  ListGroupItemText,
  Badge,
  Button
} from 'reactstrap';
import { ListGroupItem } from 'reactstrap';

const badgeButton = {
  cursor: 'pointer',
  margin: '0 0 0 5px',
  width: '22px'
};

class OrderItem extends Component {
  increase() {
    this.props.increaseCallback(this.props.index);
  }

  decrease() {
    this.props.decreaseCallback(this.props.index);
  }

  remove() {
    this.props.removeCallback(this.props.index);
  }

  render() {
    return (
      <ListGroupItem>
        <div>Product Item {this.props.order.product.id}</div>
        <ListGroupItemHeading>
          {this.props.order.product.description}
        </ListGroupItemHeading>
        <ListGroupItemText>
          <div>{this.props.order.product.price}</div>
          <div>{this.props.order.product.category}</div>
          <div>
            Quantity: {this.props.order.quantity}
            <Badge
              onClick={this.decrease.bind(this)}
              color="secondary"
              style={badgeButton}
            >
              -
            </Badge>
            <Badge
              onClick={this.increase.bind(this)}
              color="secondary"
              style={badgeButton}
            >
              +
            </Badge>
          </div>
          <div>
            Subtotal Item:{' '}
            {this.props.order.quantity *
              parseFloat(this.props.order.product.price)}
          </div>
        </ListGroupItemText>
        <Button outline color="danger" onClick={this.remove.bind(this)}>
          Remove
        </Button>
      </ListGroupItem>
    );
  }
}

export default OrderItem;
