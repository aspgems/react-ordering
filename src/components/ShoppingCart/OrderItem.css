.badgeButton {
  cursor: pointer;
  margin: 0 0 0 5px;
  width: 22px;
}
