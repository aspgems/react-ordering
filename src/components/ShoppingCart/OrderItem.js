import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ListGroupItemHeading, ListGroupItemText, Button } from 'reactstrap';
import { ListGroupItem } from 'reactstrap';
import './OrderItem.css';
import increaseItemCount from './actions';

class OrderItem extends Component {
  increase() {
    this.setState(increaseItemCount);
    // this.props.increaseCallback(this.props.index);
  }

  decrease() {
    this.props.decreaseCallback(this.props.index);
  }

  remove() {
    this.props.removeCallback(this.props.index);
  }

  total() {
    (this.props.item.quantity * this.props.item.unitPrice).toFixed(2);
  }

  render() {
    return (
      <ListGroupItem>
        <ListGroupItemHeading>
          {this.props.item.productId}{' '}
          {/* here comes the name or the description of the product */}
          <br />
        </ListGroupItemHeading>
        <ListGroupItemText>
          <Button onClick={this.decrease.bind(this)} color="secondary">
            -
          </Button>{' '}
          <Button onClick={this.increase()} color="secondary">
            +
          </Button>{' '}
          {this.props.item.quantity} x € {this.props.item.unitPrice}
        </ListGroupItemText>
        <ListGroupItemText>
          Subtotal Item: €{' '}
          {(this.props.item.quantity * this.props.item.unitPrice).toFixed(2)}
          <Button outline color="danger" onClick={this.remove.bind(this)}>
            Remove
          </Button>
        </ListGroupItemText>
      </ListGroupItem>
    );
  }
}

OrderItem.propTypes = {
  item: PropTypes.shape({
    productId: PropTypes.string.isRequired,
    unitPrice: PropTypes.number.isRequired,
    quantity: PropTypes.number.isRequired
  }),
  increaseHandle: PropTypes.func.isRequired,
  decreaseCallback: PropTypes.func.isRequired,
  removeCallback: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired
};

export default OrderItem;
