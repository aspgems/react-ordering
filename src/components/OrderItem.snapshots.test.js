import React from 'react';
import OrderItem from './OrderItem';
import renderer from 'react-test-renderer';

test('it shows correctly', () => {
  const item = {
    productId: 'A134',
    quantity: 10,
    unitPrice: 10.5,
    total: 105
  };
  const doNothing = () => {};
  const component = renderer.create(
    <OrderItem
      item={item}
      increaseItemHandler={doNothing}
      decreaseItemHandler={doNothing}
      removeItemHandler={doNothing}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
