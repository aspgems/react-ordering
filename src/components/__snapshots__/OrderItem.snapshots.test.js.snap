// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it shows correctly 1`] = `
<li
  className="list-group-item"
>
  <h5
    className="list-group-item-heading"
  >
    A134
     
    <br />
  </h5>
  <p
    className="list-group-item-text"
  >
    <button
      className="decrease-item-button btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
     
    <button
      className="increase-item-button btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
     
    10
     x € 
    10.5
  </p>
  <p
    className="list-group-item-text"
  >
    Subtotal Item: € 
    105
    <br />
    <button
      className="remove-item-button btn btn-outline-danger"
      onClick={[Function]}
      type="button"
    >
      Remove
    </button>
  </p>
</li>
`;
